### ISSUE TITLE
`allow(module_inception)` should allow specific occurrences on a case-by-case basis

### ISSUE THREAD

**Issue by crumblingstatue** [Manishearth/rust-clippy#1220]
The following does not work as expected:
```rust
mod foo {
    #[allow(module_inception)]
    mod foo {
    }
}
```

```
warning: module has the same name as its containing module, #[warn(module_inception)] on by default
 --> src/lib.rs:3:5
  |
3 |     mod foo {
  |     ^
```

Ideally, one would want to allow `module_inception` on a case-by-case basis, where one deems it acceptable.

In my case, it's [here](https://github.com/jeremyletang/rust-sfml/blob/fcc0bd9fbe2b8c351ded42a0cbb860a8d532b46e/src/window/mod.rs#L37), where `window` is the name of an SFML module that contains a variety of classes related to windowing/input, but `Window` is also a class inside that module, and I want to put each class in its own module, and `window` is the most appropriate name, since it's the class name.

Using `#![allow(module_inception)]` inside the parent module works, but then it allows `module_inception` everywhere in the module, instead of just a specific occurrence that the user would like to allow.
